test/unit/app/controllers/transactions/pending-tx-tracker-test.js

```diff
@@ -0,0 +1,616 @@
+          nonce: '0x3',
+          value: '0xfffff',
+        },
+      })
+
+      assert.ok(getCompletedTransactions.calledOnceWithExactly('0x1678a085c290ebd122dc42cba69373b5953b831d'))
+      assert.ok(!taken)
+    })
+
+    it('should return true if the nonce is taken', async function () {
+      const confirmedTxList = [{
+        id: 1,
+        hash: '0x0593ee121b92e10d63150ad08b4b8f9c7857d1bd160195ee648fb9a0f8d00eeb',
+        status: 'confirmed',
+        txParams: {
+          from: '0x1678a085c290ebd122dc42cba69373b5953b831d',
+          nonce: '0x1',
+          value: '0xfffff',
+        },
+        rawTx: '0xf86c808504a817c80082ec9d3d1d6e176d4d2593dd760e74ccac753e6a0ea0d00cc9789d0d7ff1f471d',
+      }, {
+        id: 2,
+        hash: '0x0593ee121b92e10d63150ad08b4b8f9c7857d1bd160195ee648fb9a0f8d00eeb',
+        status: 'confirmed',
+        txParams: {
+          from: '0x1678a085c290ebd122dc42cba69373b5953b831d',
+          nonce: '0x2',
+          value: '0xfffff',
+        },
+        rawTx: '0xf86c808504a817c800827b0d940c62bb760e74ccac753e6a0ea0d00cc9789d0d7ff1f471d',
+      }]
+      const getCompletedTransactions = sinon.stub().returns(confirmedTxList)
+      const pendingTxTracker = new PendingTransactionTracker({
+        query: sinon.spy(),
+        nonceTracker: {
+          getGlobalLock: sinon.stub().resolves({
+            releaseLock: sinon.spy(),
+          }),
+        },
+        getPendingTransactions: sinon.stub().returns([]),
+        getCompletedTransactions,
+        publishTransaction: sinon.spy(),
+        confirmTransaction: sinon.spy(),
+      })
+
+      const taken = await pendingTxTracker._checkIfNonceIsTaken({
+        txParams: {
+          from: '0x1678a085c290ebd122dc42cba69373b5953b831d',
+          nonce: '0x2',
+          value: '0xfffff',
+        },
+      })
+
+      assert.ok(getCompletedTransactions.calledOnceWithExactly('0x1678a085c290ebd122dc42cba69373b5953b831d'))
+      assert.ok(taken)
+    })
+  })
+
+  describe('#_checkPendingTx', function () {
+    it('should NOT emit anything if the tx is already not submitted', async function () {
+      const pendingTxTracker = new PendingTransactionTracker({
+        query: sinon.spy(),
+        nonceTracker: {
+          getGlobalLock: sinon.stub().resolves({
+            releaseLock: sinon.spy(),
+          }),
+        },
+        getPendingTransactions: sinon.stub().returns([]),
+        getCompletedTransactions: sinon.stub().returns([]),
+        publishTransaction: sinon.spy(),
+        confirmTransaction: sinon.spy(),
+      })
+      const listeners = {
+        confirmed: sinon.spy(),
+        dropped: sinon.spy(),
+        failed: sinon.spy(),
+        warning: sinon.spy(),
+      }
+
+      pendingTxTracker.once('tx:confirmed', listeners.confirmed)
+      pendingTxTracker.once('tx:dropped', listeners.dropped)
+      pendingTxTracker.once('tx:failed', listeners.failed)
+      pendingTxTracker.once('tx:warning', listeners.warning)
```

Comment:
It looks like you forgot to check that this one wasn't called

Reply:
This was missing from a few subsequent tests as well

Reply:
I'll remove this instead, this is part of the test that fails pending logic changes